File: RepoTransBench/target_projects/Python/C++/BBuf_onnx_learn/public_tests/test_public_tools_tool.cpp
#include "tools/tool.h"
#include <gtest/gtest.h>
#include <string>
#include <vector>

TEST(TestPublicToolsTool, BasicOps) {
    EXPECT_EQ(tool::add(20, 15), 35);
    EXPECT_EQ(tool::sub(30, 25), 5);
    EXPECT_EQ(tool::mul(6, 1), 6);
    EXPECT_EQ(tool::div(81, 9), 9);
}

TEST(TestPublicToolsTool, NegativeValues) {
    EXPECT_EQ(tool::sub(-10, 5), -15);
    EXPECT_EQ(tool::mul(5, -5), -25);
}

TEST(TestPublicToolsTool, DivZero) {
    EXPECT_THROW(tool::div(-10, 0), std::logic_error);
}

TEST(TestPublicToolsTool, ToolMultiply) {
    tool::Tool t;
    EXPECT_EQ(t.multiply(9, 0), 0);
}

TEST(TestPublicToolsTool, ToolIdentity) {
    tool::Tool t;
    std::vector<std::variant<int, std::string>> vec = {1, "x", 3};
    // Since C++'s identity here would just return the value itself, we test with an int
    EXPECT_EQ(t.identity(1), 1);
    // Test for a string type
    EXPECT_EQ(t.identity(std::string("x")), "x");
    // Testing an array is less trivial; simplification.
}

File: RepoTransBench/target_projects/Python/Rust/BBuf_onnx_learn/public_tests/test_public_tools_tool.rs
use bbuf_onnx_learn::tools::tool::{add, sub, mul, div, Tool};

#[test]
fn test_basic_ops() {
    assert_eq!(add(20, 15), 35);
    assert_eq!(sub(30, 25), 5);
    assert_eq!(mul(6, 1), 6);
    assert_eq!(div(81, 9), 9);
}

#[test]
fn test_negative_values() {
    assert_eq!(sub(-10, 5), -15);
    assert_eq!(mul(5, -5), -25);
}

#[test]
fn test_div_zero() {
    let result = std::panic::catch_unwind(|| { div(-10, 0); });
    assert!(result.is_err());
}

#[test]
fn test_tool_multiply() {
    let t = Tool;
    assert_eq!(t.multiply(9, 0), 0);
}

#[test]
fn test_tool_identity() {
    let t = Tool;
    assert_eq!(t.identity(vec![1, String::from("x"), 3]), vec![1, String::from("x"), 3]);
}